### PR TITLE
refactor: skip `QBFTController` assertion in `ValidatorRegistrationRunner.StartNewDuty`

### PR DESCRIPTION
**Pull request by moshe-blox**
This PR tackles #267 with a different approach, instead of passing a dummy `QBFTController` to `ValidatorRegistrationRunner`, it overrides the default behavior of `StartNewDuty` to not use the `QBFTController` at all.

### CODE REVIEW

File: ssv/validator_registration.go [hunk 1]
```diff
@@ -41,7 +42,13 @@ func NewValidatorRegistrationRunner(
 }
 
 func (r *ValidatorRegistrationRunner) StartNewDuty(duty *types.Duty) error {
-	return r.BaseRunner.baseStartNewDuty(r, duty)
+	// Note: Unlike the other runners, this doesn't call BaseRunner.baseStartNewDuty because
+	// that requires a QBFTController which ValidatorRegistrationRunner doesn't have.
+	if r.HasRunningDuty() {
+		return errors.New("already running duty")
+	}
```
Comment (GalRogozinski) [inline ssv/validator_registration.go:49]: Good addition per our conversation :-)

**Review by GalRogozinski — APPROVED** [review]

**Comment by GalRogozinski** [comment]
Fixes #270

**Comment by alonmuroch** [comment]
Why not just replace return b.QBFTController.CanStartInstance() with

if b.QBFTController != nil {
     return b.QBFTController.CanStartInstance()
}
return nil

**Comment by GalRogozinski** [comment]
@moshe-blox 
I think the problem is currently resolved due to changes in #274 
can you validate and close the PR if that is so?

**Comment by GalRogozinski** [comment]
Actually this PR is needed to resolve something else I am working on..
@alonmuroch I don't like your solution because it is error-prone. 

qbft instance shouldn't be nil for duties

**Comment by GalRogozinski** [comment]
Will be merged as part of #295

**Pull request closed without merging**